src/main/java/seedu/address/model/person/JobTitle.java

```diff
@@ -0,0 +1,58 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents an InternshipApplication's jobTitle in the tracker.
+ * Guarantees: immutable; is valid as declared in {@link #isValidJobTitle(String)}
+ */
+public class JobTitle {
```

Comment:
I'm considering whether we shall use `enum` or `class` to encapsulate the JobTitle 🤔